packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -571,9 +571,9 @@ Blockly.JavaScript.trade_definition_tradeoptions = block => {
             duration_unit      : '${duration_type}',
             currency           : '${currency}',
             amount             : ${stake_amount},
-            prediction         : ${prediction_value || 'undefined'},
-            barrierOffset      : ${barrier_offset_value || 'undefined'},
-            secondBarrierOffset: ${second_barrier_offset_value || 'undefined'},
+            ${prediction_value ? `prediction: ${prediction_value},` : ''}
+            ${barrier_offset_value ? `barrierOffset: ${barrier_offset_value},` : ''}
+            ${second_barrier_offset_value ? `secondBarrierOffset: ${second_barrier_offset_value},` : ''}
```

Comment:
@review-deriv this one looks a bit confusing as if the prediction_value or barrier_offset_value is undefined then the key should be empty string. Okay so if we just pass this like `prediction: ${prediction_value ?? undefined}` then are we facing any other issue?